src/Perl6/Metamodel/GenericHOW.nqp

```diff
@@ -22,7 +23,8 @@ class Perl6::Metamodel::GenericHOW
         my $meta := self.new();
         my $obj := nqp::settypehll(nqp::newtype($meta, 'Uninstantiable'), 'Raku');
         $meta.set_name($obj, $name);
-        $obj
+        nqp::settypecheckmode($obj,
+            nqp::const::TYPE_CHECK_NEEDS_ACCEPTS)
```

Comment:
I think `TYPE_CHECK_CACHE_DEFINITIVE` can be used here. The cache can be set to `Mu`. This would be equivalent to `accepts_type` being always true.

BTW, I memory serves me right, I was experimenting with making `GenericHOW` to implement `TypePretense`. Sure enough, I don't remember what's went wrong back then, but it still looks reasonable to me.

Reply:
Interesting. That avoids the `accepts_type` and is otherwise equivalent under most circumstances, the exception being prior `type_check`s that fail, e.g. generic-generic smartmatch. I added this because I thought signatures would need that check in particular in order to match other generic signatures' generics, but the new signature tests pass fine without.